cmake_unofficial/CMakeLists.txt

```diff
@@ -167,8 +167,8 @@ if(NOT XXHASH_BUNDLED_MODE)
   # configure and install pkg-config
   set(PREFIX ${CMAKE_INSTALL_PREFIX})
   set(EXECPREFIX "\${prefix}")
-  set(INCLUDEDIR "${CMAKE_INSTALL_INCLUDEDIR}")
-  set(LIBDIR "${CMAKE_INSTALL_LIBDIR}")
+  set(INCLUDEDIR "${CMAKE_INSTALL_FULL_INCLUDEDIR}")
```

Comment:
This is not right. You seem to be setting an absolute path. It needs to be relative to the `prefix`

Reply:
cmake allows the user to set the full path to CMAKE_INSTALL_LIBDIR (what is used in a couple of distros), so assuming the path will always be relative to prefix is not reliable with cmake.

Reply:
@ilya-fedin look at the cmake changes here https://github.com/pybind/pybind11/pull/4077/files

Reply:
or here https://github.com/HdrHistogram/HdrHistogram_c/blob/main/CMakeLists.txt

Reply:
you can pretty easily do this to support both ways. cc @eli-schwartz since you have have submitted a few PRs in this arena

Reply:
:thinking: is there any difference though? It will be the same string after all while absolute path allows just to use the builtin variable existing in cmake for a long time.

Reply:
> cmake allows the user to set absolute path to CMAKE_INSTALL_LIBDIR (what is used in a couple of distros), so assuming the path will always be relative to prefix is not reliable with cmake.

This is why you MUST use a smart path-joining function that takes CMAKE_INSTALL_LIBDIR without "FULL", checks if it is an absolute path, and:
- if absolute, uses it as-is to create an inferior pkg-config file
- if relative, joins it with `${prefix}`

Reply:
In short, please use @jtojnar's https://github.com/jtojnar/cmake-snips/

As is in fact recommended for NixOS. :p

Reply:
> :thinking: is there any difference though? It will be the same string after all while absolute path allows just to use the builtin variable existing in cmake for a long time.

That is broken in various situations, please don't break it. pkg-config needs to know the installation prefix in various cases more complicated than just running `pkgconf --cflags --libs libfoobar` on a Linux (and only Linux) system where the software is installed as a distro package.

It's the same reason that Meson handles GNU directory variables specially (guaranteeing they are always relativized even when defined as full paths, providing `join_paths()`) and has a builtin pkg-config generator (builds variables for you, supports relocatable pkg-config files).

Reply:
Thanks for explaining, done

Reply:
Great, thanks for the update. :)